Firestore/core/src/remote/firebase_metadata_provider_apple.mm

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/remote/firebase_metadata_provider_apple.h"
+
+#import "FirebaseCore/Sources/Private/FIRAppInternal.h"
+#import "FirebaseCore/Sources/Private/FIRHeartbeatInfo.h"
+#import "FirebaseCore/Sources/Private/FIROptionsInternal.h"
+
+#include "Firestore/core/src/util/string_apple.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+namespace firebase {
+namespace firestore {
+namespace remote {
+
+namespace {
+
+using util::MakeString;
+
+std::string GetUserAgent() {
+  return MakeString([FIRApp firebaseUserAgent]);
+}
+
+std::string GetHeartbeat() {
+  return std::to_string([FIRHeartbeatInfo heartbeatCodeForTag:@"fire-fst"]);
+}
+
+std::string GetGmpAppId(FIRApp* app) {
+  return MakeString(app.options.googleAppID);
+}
+
+}  // namespace
+
+FirebaseMetadataProviderApple::FirebaseMetadataProviderApple(FIRApp* app)
+    : app_(app) {
+}
+
+void FirebaseMetadataProviderApple::UpdateMetadata(
+    grpc::ClientContext& context) {
+  context.AddMetadata(kXFirebaseClientHeader, GetUserAgent());
+  context.AddMetadata(kXFirebaseClientLogTypeHeader, GetHeartbeat());
+
+  std::string gmp_app_id = GetGmpAppId(app_);
+  if (!gmp_app_id.empty()) {
+    context.AddMetadata(kXFirebaseGmpIdHeader, gmp_app_id);
+  }
```

Comment:
android does not send the gmp app id either. @VinayGuthal, curious why that is the case

Reply:
@VinayGuthal Do you know why Android doesn't send the GMP App ID? This is something we're currently planning to add.

Reply:
hmm looks like we do collect for firebase-iid. To estimate the number of devices using firebase we wouldn't need the GMP_APP_ID however. However the platform log proto has GMP_APP_ID as a optional string which can be useful for other metrics. I can change the android implementation to include the GMP_APP_ID as well.